[GPS] Add GPS#travelled=... event  (#3099)

TD-er opened the pull request:
Fixes: #3099

TD-er committed:
[GPS] Add GPS#travelled=... event  (#3099)

Fixes: #3099

TD-er committed:
Merge branch 'mega' into feature/GPS_traveled

Wiki591 commented:
Wow, that was pretty fast, I didn't expect that. A Big Thumbs Up.

TD-er commented:
> Wow, that was pretty fast, I didn't expect that. A Big Thumbs Up.

Have you tested it?

Wiki591 commented:
I am a lousy coder, I am still not able to setup a working Atom / Platformio environment. No, I have to wait until it will be included in a prebuild release.

TD-er commented:
A test build:
[ESPEasy_ESP82xx_mega-20200515-90-PR_3103.zip](https://www.dropbox.com/s/ltxhzyod41nl228/ESPEasy_ESP82xx_mega-20200515-90-PR_3103.zip?dl=0)
[ESPEasy_ESP32_mega-20200515-90-PR_3103.zip](https://www.dropbox.com/s/y1421pt4fruj8kk/ESPEasy_ESP32_mega-20200515-90-PR_3103.zip?dl=0)

TD-er committed:
Merge branch 'mega' into feature/GPS_traveled

Wiki591 commented:
You are the very best, my hero, Thank You!
In a first throw I lowered the Distance Update Interval to 2m to get some events and it seems to work fine. Lateron today I will have to move the vehicle, I will report.
Interesting effect: at the moment of changing the interval back to 100m came a GPS#travelled eventvalue of >50.000 or somewhat like that. I suppose, saving the settings sets long/lat back to zero or so. Due to the fact, that being on the road will prevend me successfully from altering settings its no problem at all, I only wanted to notice that.

TD-er commented:
> Due to the fact, that being on the road will prevend me successfully from altering settings its no problem at all, I only wanted to notice that.

Good observation.
I will have a look at it.
That's why PR's should be tested before merged :)

TD-er committed:
Merge branch 'mega' into feature/GPS_traveled

Wiki591 commented:
You've stored releases for 8266 and 32. Should I test ESP32, too? If so, I will do it, I'm only asking to know if I should order another GPS mouse.

TD-er committed:
[GPS] Add sanity check for reporting distance travelled event

TD-er commented:
In the build process I split the builds for ESP82xx and ESP32 into separate ZIP files.
This should make the flashing a bit more clear.
Also the flashing for ESP32 differs from that of ESP82xx, so that also makes it more obvious to split those.

The GPS code is the same for both, so no need to test them both.

By the way, I have now added some sanity check. if the last known position is not set, the computed distance is negative, so will not trigger an event.

Wiki591 commented:
And with the version I've today morning I can tell you: YOU MADE MY DAY!

I just made a short trip with a.distance of 30.5 km odometer reading. Reading of the trip counter: 30.46 km. On the same trip last week using the addition of 100 m every GPS event my trip counter has been 4.5 km off.

Well done, nice job, Thank You.
